### ISSUE TITLE
Error building tdebase in current

### ISSUE THREAD

**Issue by kermitdafrog8** [Ray-V/tde-slackbuilds#38]
I'm getting an implicit declaration of function pthread_setname_np in tdm/backend/dm.c:1719:9

Trying to build 14.1.2

**Comment by Ray-V** [comment]
In my Slackware installaton, pthread_setname_np is defined in pthread.h and that is included in line 59 in dm.c - if HAVE_PTHREAD_SETNAME_NP is set. Is it in your build?

That is determined in tdm/ConfigureChecks.cmake:
check_function_exists( pthread_setname_np HAVE_PTHREAD_SETNAME_NP )

It is in my builds:
...build-tdebase/config.h:#define HAVE_PTHREAD_SETNAME_NP 1
...build-tdebase/CMakeCache.txt:HAVE_PTHREAD_SETNAME_NP:INTERNAL=1

And the build log includes:
-- Looking for pthread_setname_np
-- Looking for pthread_setname_np - found

All this is from a recent build I did to check the su/sudo backend builds for tdebase.

I don't have an up-to-date current installed, so can't comment on any issues that might have brought up.

**Comment by kermitdafrog8** [comment]
I have #define HAVE_PTHREAD_SETNAME_NP 1 set in config.h and HAVE_PTHREAD_SETNAME_NP:INTERNAL=1 set in CMakeCache.txt. 

The specific error is as follow.
```/tmp/build/tmp-tdebase/tdebase-trinity-14.1.2/tdm/backend/dm.c: In function ‘SetTitle’:
/tmp/build/tmp-tdebase/tdebase-trinity-14.1.2/tdm/backend/dm.c:1719:9: error: implicit declaration of function ‘pthread_setname_np’; did you mean ‘pthread_setcanceltype’? [-Wimplicit-function-declaration]
 1719 |         pthread_setname_np(pthread_self(), name);
      |         ^~~~~~~~~~~~~~~~~~
      |         pthread_setcanceltype
```

**Comment by Ray-V** [comment]
This looks like the gcc-14 build error reported in [issue 505](https://mirror.git.trinitydesktop.org/gitea/TDE/tdebase/issues/505)

Also read more at https://gcc.gnu.org/gcc-14/porting_to.html

This has been fixed for 14.1.3 which should be released shortly.

If you want to continue with the 14.1.2 build, the patch is straightforward - [commit cd4f41e7](https://mirror.git.trinitydesktop.org/gitea/TDE/tdebase/commit/cd4f41e76b1f4710c76610a7f225d2b7fa256b79.diff)

**Comment by kermitdafrog8** [comment]
I was able to add the patch to tdebase.SlackBuild and successfully build tdebase.  Thanks.
